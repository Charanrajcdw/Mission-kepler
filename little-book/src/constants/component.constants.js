export const MODAL = {
  members: "Members",
  newBlog: "Add New Blog",
  remove: "Remove",
};

export const MEMBERS_LIST = {
  title: "Members",
  noMembers: "No members available",
};

export const BLOG_FORM = {
  title: "Add New Blog",
  button: "ADD",
  titlePlaceholder: "Name your blog",
  detailsPlaceholder: "Write Content Here ..",
  invalidValues: "Invalid title or description",
  invalidBlog: "Blog title already exists",
  blogAdded: "New Blog added successfully",
  localBlog: "local",
};

export const WARNING_MODAL = {
  title: "Unsaved changes will be lost. Are you sure to continue?",
  continue: "Continue",
  cancel: "Cancel",
};
