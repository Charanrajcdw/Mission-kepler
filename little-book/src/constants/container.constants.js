export const SIDEBAR = {
  logo: "Little Book",
  filter: "FILTER",
  noFilter: "Add blogs to show filters",
  viewMembers: "View Members",
  switchToDark: "Switch to Dark Mode",
  switchToLight: "Switch to Light Mode",
  blogs: "Blogs",
};
