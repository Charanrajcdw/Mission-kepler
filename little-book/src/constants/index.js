export * from "./url.constants";
export * from "./theme.constants";
