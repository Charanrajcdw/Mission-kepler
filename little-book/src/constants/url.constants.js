export const API_URLS = {
  blogs: "https://jsonmockserver.vercel.app/api/blogs",
  users: "https://jsonmockserver.vercel.app/api/users",
  images: "https://jsonmockserver.vercel.app/",
};
