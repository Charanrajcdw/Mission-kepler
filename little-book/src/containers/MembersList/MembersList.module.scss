@import "../../scss/fonts";

.membersListContainer {
  color: var(--primary-font-bg);

  .membersListTitle {
    font-family: $font-roboto;
    font-size: 1.25rem;
    font-weight: 400;
    margin-bottom: 28px;
  }

  .membersContainer {
    display: grid;
    grid-template-columns: repeat(auto-fill, calc(100% / 3 - 16px));
    justify-content: space-between;
  }
}
