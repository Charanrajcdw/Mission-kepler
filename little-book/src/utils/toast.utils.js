import { toast } from "react-toastify";

export const showSuccessToast = (message, currentTheme) => {
  toast.success(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};

export const showWarningToast = (message, currentTheme) => {
  toast.warn(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};
