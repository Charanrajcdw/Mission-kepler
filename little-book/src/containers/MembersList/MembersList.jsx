import { useEffect, useState } from "react";
import styles from "./MembersList.module.scss";
import Loader from "../../components/Loader/Loader";
import MembersCard from "../../components/MembersCard/MembersCard";
import { getUsers } from "../../services/blog.services";
import { MEMBERS_LIST } from "../../constants";

const MembersList = () => {
  const [members, setMembers] = useState([]);
  const [isLoaded, setIsLoaded] = useState(false);

  useEffect(() => {
    getUsers().then((users) => {
      setMembers(users);
      setIsLoaded(true);
    });
  }, []);

  const getMembersContent = () => {
    let membersContent = "";
    if (members.length > 0) {
      membersContent = members.map((member) => <MembersCard key={member.id} {...member} />);
    } else {
      membersContent = MEMBERS_LIST.noMembers;
    }
    return membersContent;
  };

  return (
    <div className={styles.membersListContainer}>
      <h2 className={styles.membersListTitle}>{MEMBERS_LIST.title}</h2>
      {isLoaded ? <div className={styles.membersContainer}>{getMembersContent()}</div> : <Loader />}
    </div>
  );
};

export default MembersList;
