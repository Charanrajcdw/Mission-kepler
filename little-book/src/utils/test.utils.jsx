import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import ThemeContextProvider from "../contexts/theme.context";
import blogSlice from "../store/slices/blogSlice";
import { THEME } from "../constants";

const customRender = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
    ...options
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const customCreateSnapshot = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return renderer.create(<ProviderWrapper>{ui}</ProviderWrapper>).toJSON();
};

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render, customCreateSnapshot as createSnapshot };
