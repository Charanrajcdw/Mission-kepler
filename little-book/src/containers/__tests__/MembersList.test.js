import { render, screen, cleanup, waitFor, waitForElementToBeRemoved } from "../../utils/test.utils";
import axiosMock from "axios";
import MembersList from "../MembersList/MembersList";
import { MEMBERS_LIST } from "../../constants";

jest.mock("../../components/MembersCard/MembersCard", () => () => <div data-testid="membersCard" />);
jest.mock("../../components/Loader/Loader", () => () => <div data-testid="loader" />);
afterEach(cleanup);

describe("Members list UI", () => {
  it("should render members list if there are members", async () => {
    axiosMock.get.mockResolvedValueOnce({
      data: [{ id: 1 }, { id: 2 }],
    });
    render(<MembersList />);
    await waitFor(() => expect(screen.getByTestId("loader")).toBeInTheDocument());
    await waitFor(() => expect(screen.queryByTestId("loader")).toBeNull());
    const membersElements = await screen.findAllByTestId("membersCard");
    expect(membersElements).toHaveLength(2);
  });

  it("should render no members content if there are no members", async () => {
    axiosMock.get.mockResolvedValueOnce({
      data: [],
    });
    render(<MembersList />);
    await waitFor(() => expect(screen.getByTestId("loader")).toBeInTheDocument());
    await waitFor(() => expect(screen.queryByTestId("loader")).toBeNull());
    const noMembersElement = await screen.findByText(MEMBERS_LIST.noMembers);
    expect(noMembersElement).toBeInTheDocument();
  });
});

describe("Members list snapshots", () => {
  it("should match members list snapshot with members", async () => {
    axiosMock.get.mockResolvedValueOnce({
      data: [{ id: 1 }, { id: 2 }],
    });
    const { asFragment } = render(<MembersList />);
    await waitForElementToBeRemoved(screen.getByTestId("loader"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("should match members list snapshot with no members", async () => {
    axiosMock.get.mockResolvedValueOnce({
      data: [],
    });
    const { asFragment } = render(<MembersList />);
    await waitForElementToBeRemoved(screen.getByTestId("loader"));
    expect(asFragment()).toMatchSnapshot();
  });
});
