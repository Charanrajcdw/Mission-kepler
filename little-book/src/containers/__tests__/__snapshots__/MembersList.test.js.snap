// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members list snapshots should match members list snapshot with members 1`] = `
<DocumentFragment>
  <div
    class="membersListContainer"
  >
    <h2
      class="membersListTitle"
    >
      Members
    </h2>
    <div
      class="membersContainer"
    >
      <div
        data-testid="membersCard"
      />
      <div
        data-testid="membersCard"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Members list snapshots should match members list snapshot with no members 1`] = `
<DocumentFragment>
  <div
    class="membersListContainer"
  >
    <h2
      class="membersListTitle"
    >
      Members
    </h2>
    <div
      class="membersContainer"
    >
      No members available
    </div>
  </div>
</DocumentFragment>
`;
